esmvalcore/preprocessor/_derive/__init__.py

```diff
@@ -96,9 +96,7 @@ def derive(cubes, short_name, long_name, units, standard_name=None):
     except Exception as exc:
         msg = (f"Derivation of variable '{short_name}' failed. If you used "
                f"the option '--skip-nonexistent' for running your recipe, "
-               f"this might be caused by missing input data for derivation "
-               f"('{short_name}' needs the variables "
-               f"{DerivedVariable().required}).")
+               f"this might be caused by missing input data for derivation")
```

Comment:
I think the original message was potentially very helpful.
Why was this shortened? Maybe it's enough to add the `()` to turn required into a call, ala
```
DerivedVariable.required()
```
? The `()` after `DerivedVariable` should not be necessary due to it being a static method.

Reply:
I also like the original one better (I was the one who wrote it :grin:), but since a few PRs `DerivedVariable.required()` needs `project` as argument, see

https://github.com/ESMValGroup/ESMValCore/blob/bb4597c338cc947990ec0c1f64e776b14a8e3370/esmvalcore/preprocessor/_derive/_baseclass.py#L10

However, since the `derive` function does not know about the `project`, we cannot use it anymore.